### PR TITLE
Deprecate pallets stubs

### PR DESCRIPTION
**Pull request by srittau**
Part of #5423

### CODE REVIEW

File: tests/check_consistent.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     {"stdlib/@python2/builtins.pyi", "stdlib/@python2/__builtin__.pyi"},
     {"stdlib/threading.pyi", "stdlib/_dummy_threading.pyi"},
 ]
+metadata_keys = {"version", "python2", "python3", "requires", "extra_description", "obsolete_since"}
```
Comment (Akuli) [inline tests/check_consistent.py:24]: What's `extra_description`? It doesn't seem to be used anywhere else in this PR.

Comment (srittau) [reply tests/check_consistent.py:24]: It's unused at the moment, but I added it in typeshed-internal/stub_uploader#11 for future use.

Comment (srittau) [reply tests/check_consistent.py:24]: I plan to document `METADATA.toml`, but I need to resolve #5422 first.

**Review by Akuli — APPROVED** [review]

**Comment by srittau** [comment]
Now let's wait and see whether it work and the deprecation text gets added to the packages.

**Comment by srittau** [comment]
🎉 Yay! https://pypi.org/project/types-click/ 🎉

**Pull request merged**